Meshtastic/MeshtasticAppDelegate.swift

```diff
@@ -27,11 +32,66 @@ class MeshtasticAppDelegate: NSObject, UIApplicationDelegate, UNUserNotification
 		}
 		return true
 	}
+	
+	func addBleManager(bleManager: BLEManager){
+		self.bleManager = bleManager
+	}
 	func userNotificationCenter(_ center: UNUserNotificationCenter, willPresent notification: UNNotification, withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
 	}
 	// This method is called when user clicked on the notification
 	func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
 		let userInfo = response.notification.request.content.userInfo
+		
+		
+		
+		if (bleManager != nil){
+			switch response.actionIdentifier {
+			case UNNotificationDefaultActionIdentifier:
+				
+				break
+				
+			case "messageNotification.thumbsUpAction":
+				let tapbackResponse = bleManager!.sendMessage(
+					message: Tapbacks.thumbsUp.emojiString,
+					toUserNum:  userInfo["userNum"] as? Int64 ?? 0,
+					channel: userInfo["channel"] as! Int32,
+					isEmoji: true,
+					replyID: userInfo["messageId"] as! Int64
+				)
+				
+				print("Tapback Response sent")
```

Comment:
See if you can migrate this to use the new logging APIs that @garthvh and I worked on instead of `print()`